pulser-core/pulser/sequence/sequence.py

```diff
@@ -364,6 +365,132 @@ def config_slm_mask(self, qubits: Iterable[QubitId]) -> None:
+        sample_seq = sample(self)
+        channel_match: dict[str, Any] = {}
+        channel_match["alert_phase_jump"] = False
+        channel_match["strict_error_message"] = ""
+        channel_match["ch_type_er_mess"] = ""
+        for o_d_ch_name, o_d_ch_obj in channel_list:
+            for n_d_ch_id, n_d_ch_obj in new_device.channels.items():
+                # Find the corresponding channel on the new device
+                # We verify the channel class then
+                # check whether the addressing Global or local
+                basis_match = o_d_ch_obj.basis == n_d_ch_obj.basis
+                addressing_match = (
+                    o_d_ch_obj.addressing == n_d_ch_obj.addressing
+                )
+                if not (basis_match and addressing_match):
+                    channel_match[o_d_ch_name] = None
+                    channel_match[
+                        "ch_type_er_mess"
+                    ] = f"No match for channel {o_d_ch_name}."
+                    continue
+                if not strict:
+                    channel_match[o_d_ch_name] = n_d_ch_id
+                    break
+                ch_samples = sample_seq.channel_samples[o_d_ch_name]
+                ch_sample_phase = ch_samples.phase
+                # Find if there is phase change between pulses or not
+                phase_is_constant = True
+                if ch_sample_phase.size != 0:
+                    phase_is_constant = bool(
+                        np.all(ch_sample_phase == ch_sample_phase[0])
+                    )
+                # Phase_jump_time and cloc_period check
+                phase_jump_time_check = (
+                    o_d_ch_obj.phase_jump_time == n_d_ch_obj.phase_jump_time
+                )
+                clock_period_check = (
+                    o_d_ch_obj.clock_period % n_d_ch_obj.clock_period == 0
+                )
+                if phase_is_constant:
+                    if clock_period_check:
+                        channel_match[o_d_ch_name] = n_d_ch_id
+                        channel_match["alert_phase_jump"] = (
+                            phase_jump_time_check is False
+                        )
+                        break
+                    else:
+                        channel_match[o_d_ch_name] = None
+                        channel_match["strict_error_message"] = (
+                            "No channel with phase_jump_time"
+                            + " & clock_period match."
+                        )
+                    continue
+                if phase_jump_time_check and clock_period_check:
+                    channel_match[o_d_ch_name] = n_d_ch_id
+                    break
+                else:
+                    channel_match[o_d_ch_name] = None
+                    channel_match["strict_error_message"] = (
+                        "No channel with phase_jump_time"
+                        + " & clock_period match."
+                    )
```

Comment:
I think there is some redundancy here. Doesn't this block do the same?

```suggestion
                if clock_period_check and (phase_is_constant or phase_jump_time_check):
                    channel_match[o_d_ch_name] = n_d_ch_id
                    channel_match["alert_phase_jump"] = (
                            not phase_jump_time_check
                        )
                    break
                channel_match[o_d_ch_name] = None
                channel_match["strict_error_message"] = (
                            "No channel with phase_jump_time"
                            + " & clock_period match."
                        )
                continue
```